shared/settings/index.js

```diff
@@ -1,20 +1,43 @@
 // @flow
+import * as React from 'react'
 import SettingsContainer from './render'
 import pausableConnect from '../util/pausable-connect'
 import {switchTo} from '../actions/route-tree'
 import {logout} from '../actions/login/creators'
 
+import type {Tab} from '../constants/settings'
+import type {TypedState} from '../constants/reducer'
 import type {RouteProps} from '../route-tree/render-route'
 
-// $FlowIssue type this connector
-export default pausableConnect(
-  (state, {routeSelected, routeLeafTags}: RouteProps<{}, {}>) => ({
-    badgeNumbers: {}, // TODO add badging logic
-    selectedTab: routeSelected,
-    isModal: routeLeafTags.modal,
-  }),
-  (dispatch, {routePath}: RouteProps<{}, {}>) => ({
-    onTabChange: tab => dispatch(switchTo(routePath.push(tab))),
-    onLogout: () => dispatch(logout()),
-  })
-)(SettingsContainer)
+type StateProps = {
+  badgeNumbers: {[key: Tab]: number},
+  selectedTab: Tab,
+}
+
+const mapStateToProps = (state: TypedState, {routeSelected}: RouteProps<{}, {}>) => ({
+  badgeNumbers: {}, // TODO add badging logic
+  // TODO: Is there a way to validate that routeSelected is a Tab?
```

Comment:
What would be the right way to fix this? Would we have to type `routeSelected` as the union of all possible route types?

Reply:
unclear. likely not worth making work now